src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is not available: " + index);
+        }
+        if (size == elementData.length) {
+            grow();
+        }
+        System.arraycopy(elementData, index, elementData, (index + 1), (size - index));
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isValidIndex(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        isValidIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isValidIndex(index);
+        T oldValue = (T) elementData[index];
+        deleteElementInArray(index);
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T equalsElement;
+        for (int i = 0; i < elementData.length; i++) {
+            T currentElement = (T) elementData[i];
+            if (equalsElement(currentElement, element)) {
+                equalsElement = element;
+                deleteElementInArray(i);
+                return equalsElement;
+            }
+        }
+        throw new NoSuchElementException("Can`t find element" + element);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public void grow() {
+        grow(size + 1);
```

Comment:
not done correctly
![image](https://github.com/mate-academy/jv-array-list/assets/109764850/eb2d9ad4-8b00-4c76-b7ac-75ef0b43947d)